Dockerfile

```diff
@@ -23,32 +23,37 @@ RUN apt-get install -y ros-crystal-test-msgs
 RUN chmod +x ./opt/ros/crystal/setup.sh
 
 # Prepare soss
-RUN mkdir -p root/soss_wp/src/osrf
-COPY ./src root/soss_wp/src/osrf #TODO: Download from git
-WORKDIR root/soss_wp
-
+RUN apt-get install -y git #Required for config
+RUN mkdir -p root/soss_wp/src
+WORKDIR /root/soss_wp/src
+RUN git clone https://github.com/osrf/soss_v2.git
+WORKDIR /root/soss_wp
 
 # Compile soss
 RUN . /opt/ros/crystal/setup.sh && \
     colcon build --packages-up-to soss-ros2-test --cmake-args -DCMAKE_BUILD_TYPE=RELEASE --install-base /opt/soss
 
+# Check compilation
+RUN . /opt/ros/crystal/setup.sh && \
+    colcon test --packages-up-to soss-ros2-test --install-base /opt/soss
+
 # Prepare environment
 WORKDIR /root
 RUN rm -rf soss_wp
-RUN mkdir -p my_wp/src
-WORKDIR /root/my_wp
+RUN mkdir -p workspace/src
+WORKDIR /root/workspace
 
 # Plugin: dds
-COPY ./src/dds src/dds #TODO: Download from git
+RUN git clone https://github.com/eProsima/SOSS-DDS.git dds
```

Comment:
I think you are cloning this into the workspace folder instead of the src folder. Will work anyway, because colcon will be able to find it there,  so leave it like that if you want